### DISCUSSION TITLE
Adding IIS External Metrics Data In Skywalking Dashboard

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#10053] [General]
Hi Team,
OAP Version : 8.9.1
I am trying to create a section for IIS metrics in Skywalking UI. Using WMI exporter i am able to fetch data and the same tried to populate in Prometheus. So either from Prometheus or directly can i pass it to the OAP backend and display in the dashboard?  Or is it possible to monitor IIS via Skywalking?

How can i add the metrics data in the dashboard? I have tried with the telemetry option.But i am not getting how can i add the metrics parameters to the OAP and get from Prometheus.

Thanks in Advance. If there are any configuration samples or methods to monitor the IIS i think that is also good for me.

**Comment by wu-sheng** [comment]
9.x provide better Prometheus support. Use Open telemetry fetches the endpoint, and forward to OAP in open census or otlp format.
Also, SkyWalking is pre-aggregation, not at query stage like using Prometheus PromQL.
So learn MAL(meter analysis language).

https://skywalking.incubator.apache.org/docs/main/next/en/concepts-and-designs/mal/

https://skywalking.incubator.apache.org/docs/main/next/en/setup/backend/opentelemetry-receiver/

**Reply by AjinThankachan** [reply]
Hi Thanks for the prompt response. Is this possible through the OAP 8.9.1. As of now i was implemented this. So for upgrade i need to get some time. If that is possible for temporary i can test and configure it. 

Also the in the database do i need to create any table for the same. Or it will automatically be created.

**Reply by wu-sheng** [reply]
If I remember correctly, you could. MAL exists for a long time. The doc should be very close.

Notice this, https://skywalking.apache.org/events/v8-eol/. With EOL, you can't find the docs on the website, the only place you could find the doc is, https://github.com/apache/skywalking/blob/v8.9.1/docs/en/concepts-and-designs/mal.md

Meanwhile, the v8 UI is very different from the v9 UI. You need to check whether it could configure metrics as you expected. There is not many docs about it.

**Reply by AjinThankachan** [reply]
Sorry to bother you once again.Is there any other option to get IIS .Net applications or IIS requests? I have tried with Promethus but not getting much data. So will Skywalking has any out of box functionality to do it? Thanks for continuous support and guidence.

**Reply by wu-sheng** [reply]
The whole .net part is not in the foundation scope, I am afraid. 
I am not familiar with dotnet part.
